google/cloud/aiplatform/utils/pipeline_utils.py

```diff
@@ -88,6 +89,10 @@ def update_runtime_parameters(
           parameter_values (Mapping[str, Any]):
               Optional. The mapping from runtime parameter names to its values.
         """
+        if parameter_values:
+            for k, v in parameter_values.items():
+                if isinstance(v, (dict, list, bool)):
+                    parameter_values[k] = json.dumps(v)
```

Comment:
The type `Mapping` doesn't require `__setitem__`: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

Recommend either qualifying the type to `MutableMapping` or copying to a dictionary instead of  updating in place. Preference for copying unless there is a reason not to.

Reply:
Fixed :)